backend/src/main/java/ca/bc/gov/app/service/client/ClientService.java

```diff
@@ -0,0 +1,31 @@
+package ca.bc.gov.app.service.client;
+
+import ca.bc.gov.app.dto.client.ClientCodeTypeDto;
+import ca.bc.gov.app.repository.client.ClientTypeCodeRepository;
+import java.time.LocalDate;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Mono;
+
+@Service
+@RequiredArgsConstructor
+public class ClientService {
+  private final ClientTypeCodeRepository clientTypeCodeRepository;
+
+  public Mono<List<ClientCodeTypeDto>> findActiveClientTypeCodes(LocalDate targetDate) {
+
+    return
+        clientTypeCodeRepository
+            .findActiveAt(targetDate)
+            .map(entity -> new ClientCodeTypeDto(
```

Comment:
@paulushcgcj it would be nice to have some repetitive code in a common place like in the CoreUtil. For example, the mapping and sorting can be placed in a public method. This used to be in the previous code. This will help to keep the code DRY.

Example:
	@Override
	public List<CodeDescrVO> toSortedCodeDescrVOs(List<? extends AbstractCodeDescrEntity> codeDescrEntities) {
		Comparator<AbstractCodeDescrEntity> comparatorByDescription = (e1, e2) ->
				(e1.getDescription()).compareTo(e2.getDescription());
		
		return codeDescrEntities.
				stream().
				sorted(comparatorByDescription).
				map(this::toCodeDescrVO).
				collect(Collectors.toList());
	}

Reply:
Indeed, but this is used only once and only here. If and when the requirements for doing this over and over appears, we will move it to some other place. For now, let's follow the KIS principle.

Reply:
The issue I have faced in the past with large groups is that people tent to "copy and paste" and follow patterns without caring much about being DRY. 

Let's please put this in a common place (same thing with the DATE request) so we just do it right from the beginning. 

Thanks.